buffer/ring_growing.go

```diff
@@ -70,3 +70,13 @@ func (r *RingGrowing) WriteOne(data interface{}) {
 	r.data[(r.readable+r.beg)%r.n] = data
 	r.readable++
 }
+
+// Len returns the number of items in the buffer.
+func (r *RingGrowing) Len() int {
+	return r.readable
+}
+
+// Capacity returns the capacity of the buffer.
+func (r *RingGrowing) Capacity() int {
```

Comment:
Sorry to nitpick, but why short `Len` and long `Capacity`?  Go's `len` and `cap` seem like they should convert to `Len` and `Cap` ?